File: src/main/java/com/github/mgramin/sqlboot/model/uri/wrappers/JsonWrapper.java
/*
 * The MIT License (MIT)
 * <p>
 * Copyright (c) 2016-2019 Maksim Gramin
 * <p>
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 * <p>
 * The above copyright notice and this permission notice shall be included
 * in all copies or substantial portions of the Software.
 * <p>
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.github.mgramin.sqlboot.model.uri.wrappers;

import com.github.mgramin.sqlboot.model.uri.Uri;
import java.util.List;
import java.util.Map;

/**
 * @author Maksim Gramin (mgramin@gmail.com)
 * @version $Id$
 * @since 0.1
 */
public class JsonWrapper implements Uri {

    private final Uri origin;

    public JsonWrapper(Uri origin) {
        this.origin = origin;
    }

    @Override
    public String type() {
        return origin.type();
    }

    @Override
    public List<String> path() {
        return origin.path();
    }

    @Override
    public String path(Integer index) {
        return origin.path(index);
    }

    @Override
    public Boolean recursive() {
        return origin.recursive();
    }

    @Override
    public Map<String, String> params() {
        return origin.params();
    }

    @Override
    public String action() {
        return origin.action();
    }

    @Override
    public String toString() {
        String s = "DbUri{" +
            "type='" + origin.type() + '\'' +
            ", path=" + origin.path() +
            ", recursive=" + origin.recursive() +
            ", params=" + origin.params() +
            "}";
        s = s.replace("%", "*");
        return s;
    }


}

File: src/main/kotlin/com/github/mgramin/sqlboot/model/uri/wrappers/JsonWrapper.kt
/*
 * BSD 3-Clause License
 *
 * Copyright (c) 2019, CROC Inc.
 * All rights reserved.
 *
 * Redistribution and use in source and binary forms, with or without
 * modification, are permitted provided that the following conditions are met:
 *
 * 1. Redistributions of source code must retain the above copyright notice, this
 *    list of conditions and the following disclaimer.
 *
 * 2. Redistributions in binary form must reproduce the above copyright notice,
 *    this list of conditions and the following disclaimer in the documentation
 *    and/or other materials provided with the distribution.
 *
 * 3. Neither the name of the copyright holder nor the names of its
 *    contributors may be used to endorse or promote products derived from
 *    this software without specific prior written permission.
 *
 * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
 * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
 * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
 * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
 * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
 * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
 * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
 * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 */

package com.github.mgramin.sqlboot.model.uri.wrappers

import com.github.mgramin.sqlboot.model.uri.Uri
import java.nio.charset.StandardCharsets

/**
 * @author Maksim Gramin (mgramin@gmail.com)
 * @version $Id: c225f8da4e169e4c72a64aeca1d201cbb46fffa0 $
 * @since 0.1
 */
class JsonWrapper(private val origin: Uri) : Uri {

    override fun type() = origin.type()

    override fun path() = origin.path()

    override fun path(index: Int) = origin.path(index)

    override fun params() = origin.params()

    override fun action() = origin.action()

    override fun connection() = origin.connection()

    override fun toString(): String {
        var s = "DbUri{" +
                "type='" + origin.type() + '\''.toString() +
                ", path=" + origin.path() +
                ", params=" + origin.params() +
                "}"
        s = java.net.URLDecoder.decode(s, StandardCharsets.UTF_8.name()).replace("%", "*")
        return s
    }
}
